m4/check_pkg_nvtx.m4

```diff
@@ -0,0 +1,72 @@
+# -*- autoconf -*-
+#
+# Copyright (c) 2024      Amazon.com, Inc. or its affiliates. All rights reserved.
+#
+# See LICENSE.txt for license information
+#
+
+AC_DEFUN([CHECK_PKG_NVTX], [
+  check_pkg_found=yes
+
+  check_pkg_CPPFLAGS_save="${CPPFLAGS}"
+  check_pkg_LDFLAGS_save="${LDFLAGS}"
+  check_pkg_LIBS_save="${LIBS}"
+
+  AC_ARG_WITH([nvtx],
+     [AS_HELP_STRING([--with-nvtx=DIR], [Enable tracing capability with NVTX @<:@default=no@:>@])])
+
+  AS_IF([test "${with_nvtx}" = "yes"],
+        [],
+        [test "${with_nvtx}" = "no"],
+        [check_pkg_found=no],
+        [AS_IF([test -d ${with_nvtx}/lib64], [check_pkg_libdir="lib64"], [check_pkg_libdir="lib"])
+         CPPFLAGS="-I${with_nvtx}/include ${CPPFLAGS}"
+         LDFLAGS="-L${with_nvtx}/${check_pkg_libdir} ${LDFLAGS}"])
+
+  AS_IF([test "${check_pkg_found}" = "yes"],
+        [AC_CHECK_LIB([nvToolsExt], [nvtxRangePop], [], [check_pkg_found=no])])
+
+  AS_IF([test "${check_pkg_found}" = "yes"],
+        [check_pkg_define=1
+         $1],
+        [check_pkg_define=0
+         CPPFLAGS="${check_pkg_CPPFLAGS_save}"
+         LDFLAGS="${check_pkg_LDFLAGS_save}"
+         LIBS="${check_pkg_LIBS_save}"
+         $2])
+
+  AC_DEFINE_UNQUOTED([HAVE_NVTX_TRACING], [${check_pkg_define}], [Defined to 1 if NVTX is available])
+
+  AS_IF([test "${check_pkg_found}" = "yes"],
+        [n_nvtx_domain_per_comm=8
+         nvtx_trace_per_comm=1
+         nvtx_trace_per_dev=0
+
+         AC_ARG_ENABLE([n_nvtx_domain_per_comm],
+                       AS_HELP_STRING([--enable-n-nvtx-domain-per-comm=N], [Number of NVTX domains to create per-communicator.
+                       For best usability, this should be equal to the max number of inflights per communicator. Default: 8]),
+                       [n_nvtx_domain_per_comm=${enableval}],
+                       [])
```

Comment:
Sorry I jumped on the review train for this late; but Nsight team's recommendation was to have a single domain for the entire library instead of a variadic setup. I'm okay with the current domain-per-comm logic until we find a better way to use categories, but we shouldn't add a new configure flag that we later drop when we figure out the right grouping mechanism... so for this specific one I'd say we go back to a #define.

Reply:
Ok